tests/test_api_v1.py

```diff
@@ -44,6 +44,40 @@ def test_create_and_fetch_request(mock_chain, client, db):
     assert created_request == fetched_request
 
 
+@mock.patch('cachito.web.api_v1.chain')
+def test_fetch_paginated_requests(mock_chain, client, db):
+
+    repo_template = 'https://github.com/release-engineering/retrodep{}.git'
+    for i in range(1, 50):
+        data = {
+            'repo': repo_template.format(i),
+            'ref': 'c50b93a32df1c9d700e3e80996845bc2e13be848',
+            'pkg_managers': ['gomod']
+        }
+
+        rv = client.post('/api/v1/requests', json=data)
+        assert rv.status_code == 201
+
+    # Sane defaults are provided
+    rv = client.get('/api/v1/requests')
+    assert rv.status_code == 200
+    response = json.loads(rv.data.decode('utf-8'))
+    fetched_requests = response['items']
+    assert len(fetched_requests) == 20
+    for repo_number, request in enumerate(fetched_requests, 1):
+        assert request['repo'] == repo_template.format(repo_number)
+
+    # per_page and page parameters are honored
+    rv = client.get('/api/v1/requests?page=2&per_page=10')
+    assert rv.status_code == 200
+    response = json.loads(rv.data.decode('utf-8'))
+    fetched_requests = response['items']
+    assert len(fetched_requests) == 10
+    for repo_number, request in enumerate(fetched_requests, 1):
+        # Add 10 because each page contains 10 items and we're processing the second page
```

Comment:
Could you just set `enumerate` to start at `11` instead?

Reply:
Yes, of course! Also, I realize that I ended up introducing the 1-index based sequence while populating test data. Changed it to 0-index based and it's a bit simpler.